src/IK/src/JointLimitsTask.cpp

```diff
@@ -0,0 +1,266 @@
+/**
+ * @file JointLimitsTask.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2023 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#include <types.h>
+#include <vector>
+
+#include <OsqpEigen/Constants.hpp>
+
+#include <BipedalLocomotion/IK/JointLimitsTask.h>
+#include <BipedalLocomotion/TextLogging/Logger.h>
+
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Model/Model.h>
+
+using namespace BipedalLocomotion::ParametersHandler;
+using namespace BipedalLocomotion::IK;
+
+bool JointLimitsTask::setKinDyn(std::shared_ptr<iDynTree::KinDynComputations> kinDyn)
+{
+    if ((kinDyn == nullptr) || (!kinDyn->isValid()))
+    {
+        log()->error("[JointLimitsTask::setKinDyn] Invalid kinDyn object.");
+        return false;
+    }
+
+    m_kinDyn = kinDyn;
+
+    // populate the limits with the one retrieved from the model
+    m_upperLimits = Eigen::VectorXd::Constant(m_kinDyn->getNrOfDegreesOfFreedom(), OsqpEigen::INFTY);
+    m_lowerLimits = Eigen::VectorXd::Constant(m_kinDyn->getNrOfDegreesOfFreedom(), -OsqpEigen::INFTY);
+
+    for (std::size_t jointIdx = 0; jointIdx < m_kinDyn->model().getNrOfJoints(); jointIdx++)
+    {
+        iDynTree::IJointConstPtr joint = m_kinDyn->model().getJoint(jointIdx);
+
+        // if the joint does not have limits skip it
+        if (!joint->hasPosLimits())
+        {
+            continue;
+        }
+
+        // for each DoF modelled by the joint get the limits
+        for (unsigned int dof = 0; dof < joint->getNrOfDOFs(); dof++)
+        {
+            if (!joint->getPosLimits(dof,
+                                     m_lowerLimits[joint->getDOFsOffset() + dof],
+                                     m_upperLimits[joint->getDOFsOffset() + dof]))
+            {
+                continue;
+            } else
+            {
+                m_isLimitConsideredForAllJoints = true;
+            }
```

Comment:
It seems that is enough that ``joint->getPosLimits`` returns true for only one joint in order to have ``m_isLimitConsideredForAllJoints = true``. Is this wanted?

Reply:
Nope

Reply:
Fixed here 2e8d668ed148b44c4f13dc2a42057d6f81b449fd thank you for noticing